src/web3-utils.js

```diff
@@ -147,10 +157,170 @@ export function formatUnits(
   if (typeof truncateToDecimalPlace === 'number') {
     const [whole = '', dec = ''] = valueBeforeCommas.split('.')
     if (dec) {
-      const truncatedDec = dec.slice(0, truncateToDecimalPlace).replace(/0*$/, '')
+      const truncatedDec = dec
+        .slice(0, truncateToDecimalPlace)
+        .replace(/0*$/, '')
       valueBeforeCommas = truncatedDec ? `${whole}.${truncatedDec}` : whole
     }
   }
 
   return commas ? EthersUtils.commify(valueBeforeCommas) : valueBeforeCommas
 }
+
+export function useTokenReserve(tokenSymbol) {
+  const [tokenReserves, setTokenReserves] = useState(null)
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    let cancelled = false
+    let retryTimer
+    async function fetchTokenReserve() {
+      try {
+        const [tokenAddress] = getKnownContract(`TOKEN_${tokenSymbol}`)
+        if (!tokenAddress) {
+          throw new Error(`Unsupported token symbol: ${tokenSymbol}`)
+        }
+        setLoading(true)
+
+        const tokenReserves = await getTokenReserves(
+          tokenAddress,
+          Number(env('CHAIN_ID'))
+        )
+
+        if (!tokenReserves) {
+          throw new Error('Could not fetch reserves')
+        }
+
+        if (!cancelled) {
+          const reserve = bigNum(
+            tokenReserves.tokenReserve.amount.toFixed(0, 1).toString()
+          )
+          setTokenReserves(reserve)
+          setLoading(false)
+        }
+      } catch (e) {
+        setLoading(false)
+        retryTimer = setTimeout(fetchTokenReserve, UNISWAP_MARKET_RETRY_EVERY)
+      }
+
+      return () => {
```

Comment:
Might be my old eyes, but this return should be lifted to be outside of this inner function, right?

Reply:
Yes, I think I did that :man_facepalming: